clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -457,120 +535,224 @@ public void onResponse(final ClientResponse response) {
             }
         }
 
-        private void handleRetriableError(Errors error, ClientResponse response) {
-            if (error == COORDINATOR_NOT_AVAILABLE ||
-                error == NOT_COORDINATOR ||
-                error == REQUEST_TIMED_OUT) {
-                coordinatorRequestManager.markCoordinatorUnknown(error.message(), response.receivedTimeMs());
+        /**
+         * Enqueue the request to be retried with exponential backoff. This will fail the request
+         * without retrying if the request timer expired.
+         */
+        @Override
+        void retry(long currentTimeMs, Throwable throwable) {
+            if (!requestTimer.isPresent() || requestTimer.get().isExpired()) {
```

Comment:
Do you ever update the requestTimer with the currentTime? This would be a good opportunity to do it.

Reply:
Agree, but not using timers anymore (moved to expirationTimeMs) so not needed.